stdlib/public/core/Policy.swift

```diff
@@ -28,6 +28,18 @@
 @_frozen
 public enum Never {}
 
+extension Never: Equatable {
+  public static func == (lhs: Never, rhs: Never) -> Bool {
+    switch (lhs, rhs) {
```

Comment:
This is cute but I would really rather this be spelled out.

Reply:
I’m not sure what you mean. 😕

Reply:
The switch is required by the compiler right now, no?

Reply:
`return true` rather than the switch.

Reply:
Oh, ha. Even did that here: https://github.com/pointfreeco/swift-prelude/blob/master/Sources/Prelude/Equatable.swift#L3 I suppose the switch is only required for `absurd`.

Reply:
I think `switch` is the better option.

`return true` is a bit misleading. It's easy to continue thinking that this code could possibly execute. (`fatalError("This is impossible. A Never value cannot be constructed")` would be better in this regard.)

`switch` educates the reader about `Never` and about `switch`. I think this works as a form of progressive disclosure. You don't need to know that this how it's implemented, but if you go looking, most people are bound to learn something.

I think the swift-evolution discussion bears this out. Several people expressed confusion initially, but walked away with a better understanding.